### ISSUE TITLE
LLVMSwift as package dependency in Xcode

### ISSUE THREAD

**Issue by qmathe** [harlanhaskins/LLVMSwift#221]
I have a Xcode 12.2 project where I would like to use LLVMSwift. I installed LLVM 9 with brew and followed Installation instructions. I then added LLVMSwift to my project with _File > Swift Packages > Add Package Dependency…_. The dependency references the latest commit on master. 

When I compile my target, I get the following error: `LLVMSwift/Sources/llvmshims/src/shim.cpp:1:10: fatal error: 'llvm-c/Core.h' file not found`.

From what I see in the output, flags defined in _pkgconfig/cllvm.pc_ are missing. It's my first time using Swift Package Manager and I don't know how it interacts with pkg-config. Here is how cllvm.pc was generated:

```
$ swift [snip]/SourcePackages/checkouts/LLVMSwift/utils/make-pkgconfig.swift 
Running /usr/bin/which llvm-config-9...
Running /usr/bin/which llvm-config...
Found llvm-config at /usr/local/opt/llvm@9/bin/llvm-config...
Running /usr/local/opt/llvm@9/bin/llvm-config --version...
LLVM version is 9.0.1
Running /usr/local/opt/llvm@9/bin/llvm-config --ldflags --libs all --system-libs...
Running /usr/local/opt/llvm@9/bin/llvm-config --cflags...
Writing pkg-config file to /usr/local/lib/pkgconfig/cllvm.pc...

Successfully wrote pkg-config file!
```

**Comment by matthewseaman** [comment]
Hi @qmathe,

Try adding the dependency using "Up to Next Minor" with version 0.7. The latest commit on `master` is part of what will eventually become 0.8 and requires LLVM 11 rather than 9.

**Comment by MaxDesiatov** [comment]
I'm getting a similar error with LLVMSwift 0.8 when building with Xcode:

```
'llvm-c/Analysis.h' file not found
```

I have LLVM 11 installed with homebrew, and pkg-config script did succeed.

**Comment by matthewseaman** [comment]
Make sure the llvm files installed by brew are in your PATH. Then rerun the pkg-config. Also see #235.

**Comment by MaxDesiatov** [comment]
I do have `export PATH="/opt/homebrew/opt/llvm@11/bin:$PATH"` in my `.zshrc`, but that doesn't seem to have any effect on Xcode. Is there anything else that needs to be done to make Xcode aware of this LLVM installation?

**Comment by henkvanderspek** [comment]
Same problem for me. Any suggestions @matthewseaman?

<img width="341" alt="Screenshot 2022-01-21 at 16 37 18" src="https://user-images.githubusercontent.com/1003985/150555825-8b2a1bc0-2fc5-4c98-b843-b726e7094c94.png">

**Comment by MaxDesiatov** [comment]
Somehow rerunning `swift utils/make-pkgconfig.swift` did help in my case after all.

**Comment by henkvanderspek** [comment]
Still same problem. I must be doing something different.

<img width="621" alt="Screenshot 2022-01-21 at 16 59 09" src="https://user-images.githubusercontent.com/1003985/150559278-3fe8511e-af2b-4e31-8c32-f2780caad889.png">

<img width="389" alt="Screenshot 2022-01-21 at 16 59 28" src="https://user-images.githubusercontent.com/1003985/150559286-da0a4194-b661-4eec-8d5a-fef2323cfd0e.png">

**Comment by henkvanderspek** [comment]
Do i need to use LLVM 11 instead?

**Comment by MaxDesiatov** [comment]
Also check that you don't have two version of LLVM installed at the same time, I guess some of my issues could've been cause by having `llvm@12` and `llvm@13` installed by Homebrew at the same time. Also, you have to run `make-pkgconfig.swift` every time after Homebrew installs or remove any of these packages, I think. I'm still not 100% sure what exactly worked in my case, but at one point I reran the script and Xcode builds started working again.

**Comment by MaxDesiatov** [comment]
`master` branch and 0.8.0 release need LLVM 11, while https://github.com/llvm-swift/LLVMSwift/pull/238 somewhat works with LLVM 13, but with disabled JIT.

**Comment by henkvanderspek** [comment]
Ok thanks, Max. I didn't have LLVM installed before using this package. It's a new machine and didn't even had Homebrew on it 😄

**Comment by MaxDesiatov** [comment]
As far as I understand `make-pkgconfig.swift` relies on Homebrew installations, but I could be wrong.

**Comment by henkvanderspek** [comment]
Ok, I need JIT. So I better use LLVM 11 then, remove 13 to make sure.

**Comment by henkvanderspek** [comment]
> As far as I understand `make-pkgconfig.swift` relies on Homebrew installations, but I could be wrong.

Yes, I see it uses brew. Script runs fine though. But it must be because I installed LLVM 13 instead. Still weird it complains about missing headers and doesn't even pass compilation.

**Comment by MaxDesiatov** [comment]
Good way to diagnose could be by using `swift build`. If `swift build` works, but building from Xcode doesn't, then it's the same issue I had 🙂

**Comment by henkvanderspek** [comment]
```Running /usr/bin/which brew...
Running /opt/homebrew/bin/brew --prefix...
Running /usr/bin/which llvm-config-11...
Running /usr/bin/which llvm-config...
Found llvm-config at /opt/homebrew/opt/llvm@11/bin/llvm-config...
Running /opt/homebrew/opt/llvm@11/bin/llvm-config --version...
LLVM version is 11.1.0
Running /opt/homebrew/opt/llvm@11/bin/llvm-config --ldflags --libs all --system-libs...
Running /opt/homebrew/opt/llvm@11/bin/llvm-config --cflags...
Writing pkg-config file to /opt/homebrew/lib/pkgconfig/cllvm.pc...

Successfully wrote pkg-config file!
Make sure to re-run this script when you update LLVM.
henk@Henks-MacBook-Pro LLVMSwift % swift build                     
warning: failed to retrieve search paths with pkg-config; maybe pkg-config is not installed
warning: you may be able to install cllvm using your system-packager:
    brew install llvm

/Users/henk/Library/Developer/Xcode/DerivedData/8-Bit_OS-eprcdyqouebnfjaxybvmskdpufrh/SourcePackages/checkouts/LLVMSwift/Sources/llvmshims/src/shim.cpp:1:10: fatal error: 'llvm-c/Core.h' file not found
#include "llvm-c/Core.h"
         ^~~~~~~~~~~~~~~
1 error generated.
[0/1] Compiling shim.cpp```

**Comment by henkvanderspek** [comment]
It does give more info. But still strange.

**Comment by henkvanderspek** [comment]
Oh wait. It said so to install pkg-config. That was it! Thanks for the help, Max!

**Comment by lucfis** [comment]
I'm having a very similar issue. Upon running `swift build` I get:
 `fatal error: 'llvm-c/Core.h' file not found`
`#include "llvm-c/Core.h"`
`         ^~~~~~~~~~~~~~~`
`1 error generated.`
`[0/2] Compiling shim.cpp`

As far as I know a pkg-config file has been successfully created, and I've tried the above with both LLVM 15 and (now) LLVM 11. 

I've very very new to LLVM and Swift, so it's possible I've messed up in some obvious way and am just being dumb. Are there any recommendations of things I could try?
